### ISSUE TITLE
geeqie hangs when opening a third window

### ISSUE THREAD

**Issue by geeqie-dave** [BestImageViewer/geeqie#1519]
**Setup (please complete the following information):**
 - Distribution: fedora
 - Distribution release: 37
 - Geeqie version: 2.0.1 GTK3

**Describe the bug**
I can open only two windows (full, default configuration with thumbnail view, directory tree,  and image view panes).  When I try to open a third such window (either from command line, via file browser or geeqie's window menu), the program hangs with CPU usage at 100%. 

**To reproduce**
Open geeqie.  From the window menu open a second window.  This works as expected.  Again, from the window menu, open a third window.  Nothing happens and the formerly functional two windows cease to respond.  Opening the first and second windows can be performed from command line or clicking an image file.  For the third window attempt, all of these methods result in the same hung state with the single geeqie process using 100% CPU.

**Expected behavior**
Either a third window opens or some error message or something besides a hung process.

**Screenshots**
Screen cap of my default window configuration
![geeqie_window](https://github.com/user-attachments/assets/2ee794dd-40d9-4806-b5fc-e504c4b72884)

**Image sources**
Standard fedora 37 package

**Error logs or seg. fault files**
Strace output including about 3 seconds after program hangs:
[strace.zip](https://github.com/user-attachments/files/16938661/strace.zip)

**Additional context**
I have 3 fedora 37 systems, all have the same issue.

**Comment by geeqie-dave** [comment]
Forgot to mention:  KDE Plasma desktop

**Comment by geeqie-dave** [comment]
I should have tried this before posting:  I renamed my ~/.conf/geeqie folder and can now open as many windows as I want.  I will start adding things back in and see if the problem returns.

**Comment by geeqie-dave** [comment]
Problem identified:  if I turn on folder setting "view as tree", I can not open a third window.

**Comment by qarkai** [comment]
@geeqie-dave third window with folder setting "view as tree" works for me on master. Have you tried latest releases https://github.com/BestImageViewer/geeqie/releases/tag/v2.4 and https://github.com/BestImageViewer/geeqie/releases/tag/continuous?

**Comment by caclark** [comment]
I cannot replicate this problem. Please try:

 Run geeqie each time from the command line in a new terminal window, and use `geeqie --debug=4`

Maybe the debug output will provide information.

**Comment by geeqie-dave** [comment]
No, I try to keep my distro's packages intact.  I could try building from source and running independently, but I fear the libraries may be a factor, as well, especially the GTK-based file/folder parts (I run KDE Plasma).

Dave D.
________________________________

**Comment by geeqie-dave** [comment]
No output in any of the 3 consoles, program hangs on third instance.  Turn off the tree view and no problems (except I can't see the tree view 😉).  Are you running KDE/Plasma?  I will try running a Gnome session tomorrow and see if it behaves the same.

Dave D.
________________________________

**Comment by caclark** [comment]
> No, I try to keep my distro's packages intact.  I could try building from source and running independently, but I fear the libraries may be a factor, as well, especially the GTK-based file/folder parts (I run KDE Plasma).

I understand concern people have about not wanting to change anything in system that works.

However in respect to the suggestion of using an AppImage, the AppImage is a file that is self-contained. It contains all the dependencies that is required. You download it, make it executable, run it, delete it, and nothing has changed on your system.

It is a large file, so load time is several seconds. But if you wish to take a look at what the current state of Geeqie is, an AppImage is a safe way to do that without the problems of compiling from sources.

**Comment by geeqie-dave** [comment]
The tree view setting must be applied to the default layout for the problem to manifest.  My workaround is simply to save a default layout with list view, then turn on tree view as needed.  Once the tree view is turned on, I can open new windows from the window menu and specify "from current" and I get all the new windows I want with tree views.  Only when the tree view is saved as the default layout will the third window hang the app, and then only when opening new windows from command line or file manager.  As long as default folder view is "view as list", everything works, regardless of the state of the open windows.

When  starting geeqie from command line with --debug=4, there is no output to the console window at all, even when the program hangs.

I tried a Gnome session (as opposed to my normal Plasma) and had the exact same results.  I also tried Wayland (as opposed to Xorg that I usually run) and it made no difference (Gnome on Xorg, Gnome on Wayland, Plasma on Xorg and Plasma on Wayland all behave the same).

I tried looking at the strace output (attached to my original bug report) but it is overwhelming and I got nowhere.  I will use my workaround and continue using "the best image viewer" and all the plugins I have written for it that let me manage all the thousands of pictures my wife and I have taken all over the world.  Thanks to all who have contributed.

**Comment by geeqie-dave** [comment]
I have no "Menu Style" setting(s), but I have a checkbox for "Expand menu and toolbar", which was unchecked.  I checked it and restarted geeqie:  no change.  Is this because I am not running the current version?  Since I am NOT running the current version, perhaps we are wasting our time.  I am content with leaving my default at list view and turning on tree view as needed.  I will upgrade my OS one of these days and will likely get a newer version, then, that may well solve the issue.

Dave D.

________________________________

**Comment by geeqie-dave** [comment]
Just a follow-up:  I am now running the fedora 40 version of geeqie on my fedora 37 system (downloaded fedora 40 source RPM and deps, then built for fedora 37) and it got me up to version 2.2 which still hangs on third window open attempt when  default view is set to tree.

**Comment by geeqie-dave** [comment]
Good news:  I finally got around to upgrading to fedora 40 and I can now leave my default window configuration set to tree view!  No hang, no matter how many windows I open!  Yea!  :)

Bad news:  Video thumbnails no longer work.  :(

**Comment by geeqie-dave** [comment]
I just updated my bug report about geeqie hanging when a third window was opened with folders in tree view:  problem is gone after upgrading fedora.  That's the good news.

Bad news is that video thumbnails stopped working.  Is there a way to tell if the fedora build has them enabled without downloading their build files?  I have ffmpegthumbnailer installed, but I don't see an ffmpegthumbnailer lib package being available.  Here's a list of installed packages that include "thumbnail":

totem-video-thumbnailer-43.1-1.fc40.x86_64
evince-thumbnailer-46.3.1-1.fc40.x86_64
kdegraphics-thumbnailers-24.12.2-1.fc40.x86_64
ffmpegthumbnailer-2.2.3-0.7.20240104git1b5a779.fc40.x86_64

Is geeqie depending on something else?  Is ffmpeg depending on something else that I need to install?

Dave D.

**Comment by qarkai** [comment]
> Is there a way to tell if the fedora build has them enabled without downloading their build files? 

Run `ldd $(command -v geeqie) | grep thumb`.

**Comment by geeqie-dave** [comment]
Well, looks like I'm building from source - one of these days...

Would the grep command have returned ffmpegthumbnailer, or are there other options for video thumbnails?

Thanks.

________________________________

**Comment by qarkai** [comment]
Just ffmpegthumbnailer for me:
```
$ ldd $(command -v geeqie) | grep thumb
	libffmpegthumbnailer.so.4 => /usr/lib/libffmpegthumbnailer.so.4 (0x00007ae4e50fc000)
```
